services/user/Supervisor/ui/src/component-loading/shims/ts/wasi-keyvalue.ts

```diff
@@ -158,7 +163,10 @@ class Bucket {
         }
 
         try {
-            localStorage.setItem(`${this.bucketId}:${key}`, value.toString());
+            localStorage.setItem(
+                `${this.bucketId}:${key}`,
+                btoa(decoder.decode(value)),
```

Comment:
Unfortunately `atob` and `btoa` are weird. Decoding the input as utf-8 is wrong. In particular, it will fail if `value` is not valid utf-8. The correct way to use `btoa` is to convert each element of `value` to a single code point.

Reply:
Ok, I think I got it. 
Third time's the charm 🤞